src/org/usfirst/frc4904/robot/commands/MatchInformer.java

```diff
@@ -0,0 +1,40 @@
+package org.usfirst.frc4904.robot.commands;
+
+
+import org.usfirst.frc4904.robot.RobotMap;
+import edu.wpi.first.wpilibj.DriverStation;
+
+public class MatchInformer extends CANInformer {
+	public static final int NUMBER_OF_MESSAGES = 200;
+
+	public MatchInformer() {
+		super(RobotMap.Port.CAN.matchConfigBroadcast, MatchInformer.NUMBER_OF_MESSAGES);
+	}
+
+	@Override
+	protected byte[] getCANMessage() {
+		// Game time will generally be in the 0-150 second range.
+		// All Java bytes are signed, so anything >127 will overflow into the negatives.
+		// However, the Teensy can read them as unsigned, so we're all good.
+		long roundedGameTime = Math.round(DriverStation.getInstance().getMatchTime());
+		byte gameTimeA = (byte) ((byte) roundedGameTime >> 4);
+		byte gameTimeB = (byte) ((byte) roundedGameTime & 0b1111);
+		byte gameMode = -1; // default/unknown
+		if (DriverStation.getInstance().isDisabled()) {
+			gameMode = 0;
+		}
+		if (DriverStation.getInstance().isAutonomous()) {
+			gameMode = 1;
+		}
+		if (DriverStation.getInstance().isEnabled()) {
```

Comment:
What if the robot is Enabled in Autonomous? `IsDisabled` is implemented as `!isEnabled()`, so this logic will fail. I think you want `isOperatorControl()` (which is implemented as `!(isAutonomous() || isTest())`). Also if I remember correctly `isAutonomous()` and co might return true even if auton is selected in the DS but disabled, so we should test and maybe add some `else if`s (with precedence for `isDisabled()`).